Sources/BrowserServicesKit/Common/Extensions/URLExtension.swift

```diff
@@ -121,6 +121,7 @@ extension URL {
         }
         
         var percentEncodedQueryItems = components.percentEncodedQueryItems ?? [URLQueryItem]()
+        percentEncodedQueryItems.removeAll(where: { $0.name == percentEncodedName })
```

Comment:
I suggest not to add this line that will cause replacing the parameter with the same name (is existing). Allowing multiple parameters with the same name is more aligned with the RFC specs and how `URLComponents` with `URLQueryItems` work.

Before removing that we also need to make sure if this won't be breaking change to iOS and need to review all the `addParameter` call use cases.

Reply:
I‘ve checked again against our codebase and this change is based on the existing iOS code: [URLExtension.swift:L45](https://github.com/duckduckgo/iOS/blob/9e5d8fd3a6e1adcf6d434dda64aa8c8819d6b716/Core/URLExtension.swift#L45); [URLExtension.swift:L72](https://github.com/duckduckgo/iOS/blob/9e5d8fd3a6e1adcf6d434dda64aa8c8819d6b716/Core/URLExtension.swift#L72) which is removing all matching parameters when adding one
Also the function is only used for forming URLs for the search or for our API requests so this does not affect user browsing experience and URLs entered in the address bar - those are fixed separately